### ISSUE TITLE
Javascript code in credentials?

### ISSUE THREAD

**Issue by hunhejj** [huginn/huginn#2359]
Hi, how is it possible in a JavascriptAgent to use javascript code stored in a credential (an object, functions, etc.)? `this.credential()` returns a string only. Or what is the javascript "mode" of a credential designed for?

**Comment by dsander** [comment]
I don't think that is possible. Fetching the code with `this.credential('name')` and then calling `eval` might work, but I did not try it.

The Javascript mode in credential is a legacy leftover and should probably be removed. https://github.com/huginn/huginn/issues/1856#issuecomment-270554840

**Comment by hunhejj** [comment]
eval is not too nice, but it did work, thanks. The documentation of the JavaScript agent tells this:
> You can put code in the code option, or put your code in a Credential and reference it from code with credential:<name> (recommended).
As I understand, in that case you put the entire code into the credential without the possibility to pass any custom parameter to it or set a global option before. So basically absolutely no possibility of customization. Why is that the recommended way then?

**Comment by dsander** [comment]
I did not even know about the special treatment of `credential:`. I agree that in the current form is is not super useful having the ability to mix and match code in the JavascriptAgent with snippets from credentials would probably make more sense.
